metrics-sampler/src/onboarding_metrics_sampler.rs

```diff
@@ -0,0 +1,131 @@
+//! Samples onboarding-related metrics at a fixed interval
+
+use std::time::Duration;
+
+use common::types::tasks::{HistoricalTaskDescription, WalletUpdateType};
+use state::State;
+
+use crate::sampler::AsyncMetricSampler;
+
+/// The name of the sampler for onboarding metrics
+const ONBOARDING_METRICS_SAMPLER_NAME: &str = "onboarding-metrics-sampler";
+/// The interval at which to sample onboarding metrics
+const ONBOARDING_METRICS_SAMPLE_INTERVAL_MS: u64 = 10_000;
```

Comment:
can we increase this to 10m-1h or something higher? doing a full scan of the db every 10s will not scale.

similarly this whole approach will become a performance issue as the number of wallets grow. I'm a fan of getting _something_ out the door, but i do think we should seek to replace this quite quickly.